json/src/main/scala/fs2/data/json/jsonpath/internals/JsonQueryPipe.scala

```diff
@@ -38,12 +40,17 @@ private[jsonpath] object PathMatcher {
     def intersection(that: Range): Range =
       Range(Math.max(this.low, that.low), Math.min(this.high, that.high))
 
+    def union(that: Range): Range =
+      Range(math.min(this.low, that.low), math.max(this.high, that.high))
```

Comment:
Is it somehow guaranteed that the ranges don't have a gap between them? Because if they had, their union would be different.

Example: `union([1-4], [8-13]) != [1-13]`

Reply:
Yes this is guaranteed at call site, it should be private anyway.